packages/common/src/intl/intlHelpers.ts

```diff
@@ -0,0 +1,48 @@
+import { useIntl } from 'react-intl';
+import type { PrimitiveType } from 'intl-messageformat';
+
+// "import type" ensures en messages aren't bundled by default
+import * as sourceOfTruth from './locales/en.json';
+
+export type SupportedLocales = 'en' | 'fr' | 'pt';
+export type LocaleMessages = typeof sourceOfTruth;

```

Comment:
really like this idea - of taking the keys from the main language file and checking all usages against this set of allowable keys